autogluon/task/text_prediction/models/basic_v1.py

```diff
@@ -0,0 +1,911 @@
+        if scheduler == 'fifo':
+            if searcher is None:
+                searcher = 'random'
+            scheduler = FIFOScheduler(train_fn,
+                                      time_out=time_limits,
+                                      num_trials=num_trials,
+                                      resource=resource,
+                                      searcher=searcher,
+                                      checkpoint=None,
+                                      reward_attr='reward',
+                                      time_attr='time_spent')
+        elif scheduler == 'hyperband':
+            if searcher is None:
+                searcher = 'random'
+            if grace_period is None:
+                grace_period = 1
+            if max_t is None:
+                max_t = 5
+            scheduler = HyperbandScheduler(train_fn,
+                                           time_out=time_limits,
+                                           max_t=max_t,
+                                           resource=resource,
+                                           searcher=searcher,
+                                           grace_period=grace_period,
+                                           reduction_factor=reduction_factor,
+                                           brackets=brackets,
+                                           checkpoint=None,
+                                           reward_attr='reward',
+                                           time_attr='report_idx')
+        else:
+            raise NotImplementedError
+        scheduler.run()
+        scheduler.join_jobs()
+        self._logger.info('Best_config={}'.format(scheduler.get_best_config()))
+        best_task_id = scheduler.get_best_task_id()
+        best_model_saved_dir_path = os.path.join(self._output_directory,
+                                                 'task{}'.format(best_task_id))
+        best_cfg_path = os.path.join(best_model_saved_dir_path, 'cfg.yml')
+        cfg = self.base_config.clone_merge(best_cfg_path)
+        self._results = dict()
+        self._results.update(best_reward=scheduler.get_best_reward(),
+                             best_config=scheduler.get_best_config(),
+                             total_time=time.time() - start_tick,
+                             metadata=scheduler.metadata,
+                             training_history=scheduler.training_history,
+                             config_history=scheduler.config_history,
+                             reward_attr=scheduler._reward_attr,
+                             config=cfg)
+        if plot_results:
+            plot_training_curves = os.path.join(self._output_directory, 'plot_training_curves.png')
+            scheduler.get_training_curves(filename=plot_training_curves, plot=plot_results,
+                                          use_legend=True)
+        # Consider to move this to a separate predictor
+        self._config = cfg
```

Comment:
Why set self._config here? Why not later or earlier? You set self._config and then continue to use cfg instead. Why have cfg at all, could you just set self._config to begin with?

Reply:
This is because I'm mixing the functionality of `Model` and `Predictor`. Ideally, I should create a separate Predictor object that just does the prediction jobs. I decide to refactor it later.